azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/ApplicationGatewaysImpl.java

```diff
@@ -45,4 +53,71 @@ protected ApplicationGatewayImpl wrapModel(String name) {
     protected ApplicationGatewayImpl wrapModel(ApplicationGatewayInner inner) {
         return (inner == null) ? null : new ApplicationGatewayImpl(inner.name(), inner, this.manager());
     }
+
+    @Override
+    public void start(String... applicationGatewayResourceId) {
+        if (applicationGatewayResourceId == null) {
+            return;
+        }
+        this.startAsync(applicationGatewayResourceId).toBlocking().last();
+    }
+
+    @Override
+    public Observable<String> startAsync(String...applicationGatewayResourceId) {
+        return this.startAsync(new ArrayList<String>(Arrays.asList(applicationGatewayResourceId)));
+    }
+
+    @Override
+    public void stop(String... applicationGatewayResourceIds) {
+        if (applicationGatewayResourceIds == null) {
+            return;
+        }
+        this.stopAsync(applicationGatewayResourceIds);
+    }
+
+    @Override
+    public void start(Collection<String> applicationGatewayResourceIds) {
+        this.startAsync(applicationGatewayResourceIds).toBlocking().last();
+    }
+
+    @Override
+    public void stop(Collection<String> applicationGatewayResourceIds) {
+        this.stopAsync(applicationGatewayResourceIds).toBlocking().last();
+    }
+
+    @Override
+    public Observable<String> stopAsync(String...applicationGatewayResourceIds) {
+        return this.stopAsync(new ArrayList<String>(Arrays.asList(applicationGatewayResourceIds)));
+    }
+
+    @Override
+    public Observable<String> startAsync(Collection<String> applicationGatewayResourceIds) {
+        if (applicationGatewayResourceIds == null) {
+            return null;
```

Comment:
we should return `Observable.empty()` instead of null, with null it can break consumer's stream chaining.